src/eliminate_data_type.cpp

```diff
@@ -42,31 +108,36 @@ void eliminate_data_type::apply(module& m) const
                                                            "scatternd_add",
                                                            "scatternd_mul",
                                                            "scatternd_none"};
-    for(auto ins : iterator_for(m))
+    if(unsupported_types.empty() and unsupported_fp8_ops.empty())
+    {
+        return;
+    }
+    else if(not unsupported_fp8_ops.empty() and not unsupported_types.empty())
+    {
+        MIGRAPHX_THROW("eliminate_data_type: specify either unsupported FP8 ops or unsupported "
+                       "data types not both.");
```

Comment:
This should not be an error. You should be able to pass in the data types you want to eliminate and for which operators you want to eliminate them for.

Reply:
Done.